ci/generate-docs.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+set -e -u -x -o pipefail
+
+if [ ! "$#" -eq 1 ]; then
+    echo "Expected usage: generate-docs.sh <git_hash>"
+    exit 1
+fi
+
+TAG="${1}"
+
+function cleanUp() {
+    # Ensure we are not in the temp dir before cleaning it
+    cd "${CURRENT_DIR}"
+    if [[ -n "${TMP_DIR-}" ]]; then
+        rm -rf "${TMP_DIR}"
+    fi
+}
+
+trap cleanUp EXIT
+
+cd "$(dirname "$0")/../"
+
+ci/bootstrap.sh
+
+# TODO: Remove when merged or moved to other repo.
+pushd .shared-ci
+    git checkout feature/api-docs-generator
+popd
+
+# Make a copy of this repo and the docs repo.
+CLONE_URL="git@github.com:spatialos/gdk-for-unity.git"
+
+CURRENT_DIR=$(pwd)
+TMP_DIR=$(mktemp -d)
+CODE_DIR="${TMP_DIR}/code"
+DOCS_DIR="${TMP_DIR}/docs"
+
+# Clone and checkout correct code tag.
+git clone "${CLONE_URL}" "${CODE_DIR}"
+pushd "${CODE_DIR}"
+    git checkout "${TAG}"
+popd
+
+# Clone and create branch for docs
+DOCS_BRANCH="docs/api-docs-${TAG}"
+git clone "${CLONE_URL}" "${DOCS_DIR}"
+pushd "${DOCS_DIR}"
+    git checkout docs/improbadoc-conversion
+    git checkout -b "${DOCS_BRANCH}"
+popd
+
+# Generate API docs
+dotnet run -p .shared-ci/tools/Docgen/Docgen.csproj -- \
+    --target-directory="${CODE_DIR}/workers/unity/Packages/" \
+    --target-namespace="Improbable.Gdk" \
+    --output-directory="${DOCS_DIR}/docs/api" \
+    --namespace-filter=".*EditmodeTests" \
+    --namespace-filter=".*PlaymodeTests" \
+    --api-path="api" \
+    --git-tag="${TAG}"
+
+# Commit and push
+pushd "${DOCS_DIR}"
+    git add --all
+    git commit -m 'api docs'
+    git push origin "${DOCS_BRANCH}"
```

Comment:
What happens if someone else has run this script and pushed a branch with the same tag?

Reply:
The `git push` would fail - can check if the branch already exists in the remote and print an error message